File: Jack.Extensions2/Idade.vb
Public Structure Idade

    Public ReadOnly Property Dias() As Integer
    Public ReadOnly Property Meses() As Integer
    Public ReadOnly Property Anos() As Integer

    Public Sub New(dataNascimento As DateTime)
        Me.New()
        If dataNascimento > DateTime.Now Then
            Return
        End If

        Dim hoje As DateTime = DateTime.Today

        Dim dias As Integer = hoje.Day - dataNascimento.Day
        If dias < 0 Then
            hoje = hoje.AddMonths(-1)
            dias += DateTime.DaysInMonth(hoje.Year, hoje.Month)
        End If

        Dim meses As Integer = hoje.Month - dataNascimento.Month
        If meses < 0 Then
            hoje = hoje.AddYears(-1)
            meses += 12
        End If

        Dim anos As Integer = hoje.Year - dataNascimento.Year

        Me.Dias = dias
        Me.Meses = meses
        Me.Anos = anos

    End Sub
End Structure

File: Jack.Model/Helpers/Idade.cs
using System;

namespace Jack.Model.Helpers
{
    public class Idade
    {
        public int Dias { get; }
        public int Meses { get; }
        public int Anos { get; }

        public Idade()
        {

        }
        public Idade(DateTime dataNascimento) : this(dataNascimento, DateTime.Now)
        {

        }
        public Idade(DateTime dataNascimento, DateTime dataDesejada) : this()
        {
            if (dataNascimento > dataDesejada)
            {
                return;
            }

            int dias = dataDesejada.Day - dataNascimento.Day;
            if (dias < 0)
            {
                dataDesejada = dataDesejada.AddMonths(-1);
                dias += DateTime.DaysInMonth(dataDesejada.Year, dataDesejada.Month);
            }

            int meses = dataDesejada.Month - dataNascimento.Month;
            if (meses < 0)
            {
                dataDesejada = dataDesejada.AddYears(-1);
                meses += 12;
            }

            int anos = dataDesejada.Year - dataNascimento.Year;

            Dias = dias;
            Meses = meses;
            Anos = anos;
        }

    }
}
